subsys_server/src/server.c

```diff
@@ -0,0 +1,61 @@
+#include <pthread.h>
+
+#include "client_api.h"
+
+/** @note Before refactor, this was >#define name tag<
+ * but I couldn't find a symbol for tag so I quoted as a string litteral
+ * - Carl
+ * */
+#define name "tag"
+
+static int   server_init(void);
+static void *receiving(void *);
+static void  formatting(buffer, fmt_buffer);
+static void  do_msn(fmt_buffer);
+
+
+// server.c
+
+int main()
+{
+    if (server_init() == -1)
+    {
+        fprintf(stderr, "server_init failed\n");
+    }
+    return 0;
+}
+
+
+static int server_init(void)
+{
+    if (ipc_connect(name) == -1) // ipc_connect()
+    {
+        fprintf(stderr, "ipc_connect() failed\n");
+        return -1;
+    }
+    pthread_t await_msn;
+    pthread_create(&await_msn, NULL, receiving, NULL);
+}
+
+static void *receiving(void * args)
+{
+    char buffer[MAX_MSG_LEN];
+    char fmt_buffer[MAX_MSG_LEN];
+    for (;;)
+    {
+        ipc_recv("*", buffer, MAX_MSG_LEN);
+        formatting(buffer, fmt_buffer);
+        do_msn(fmt_buffer);
+    }
+}
+
+static void formatting(buffer, fmt_buffer)
```

Comment:
Okay, so here is the big issue: C code CANNOT call C++ code. The ABI is not compatible. I'm not even sure how this was working before to be honest.

I've commented it out for now, but I'm super interested to know if I'm missing something @amellalalex

It's also possible that this is legacy code and unused in IPC

Reply:
yes you're right about that. this code was actually just being written as pseudo-code during a discussion I had with yann so that we could go over the basic source configuration of a subsystem server